reactor-core/src/main/java/reactor/core/publisher/FluxSwitchOnFirst.java

```diff
@@ -147,36 +141,38 @@ public void cancel() {
         public void onSubscribe(Subscription s) {
             if (Operators.validate(this.s, s)) {
                 this.s = s;
-                s.request(1);
+                this.outer.sendSubscription();
+                if (this.inner != Operators.emptySubscriber()) {
```

Comment:
double-check here in order to ensure that we really need to request the first element. if the subscription has been canceled already then noops required

Reply:
again, you can use `Operators.EMPTY_SUBSCRIBER` to be 100% shielded from API implementation details